engine/classes/Elgg/Filesystem/Adapter/AwsS3.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Elgg\Filesystem\Adapter;
+use Gaufrette\Filesystem;
+use Gaufrette\File;
+use Gaufrette\Adapter\AwsS3 as GaufretteAwsS3;
+use Aws\S3\S3Client;
+
+class AwsS3 extends Filesystem implements Adapter {
```

Comment:
I've grown to treat inheritance with suspicion. Is that really what we mean to do here?